### PR TITLE
[recipe] fix: prime compatibility issues

### PR DESCRIPTION
**Pull request by lsy641**
### What does this PR do?

> Fix PRIME compatibility issues, including async rollout, reward model handeling, and batch data format issue.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

1. pass pre-commit run
2. successfully sh run_prime_qwen.sh

### API and Usage Example

> N/A

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

https://github.com/volcengine/verl/issues/2547

* recipe/prime/main_prime.py
Trainer now picks AsyncActorRolloutRefWorker automatically when actor_rollout_ref.rollout.mode=async, keeping sync behavior unchanged but restoring async rollouts.
* recipe/prime/prime_ray_trainer.py
compute_response_mask now returns a DataProto, so batch.union(response_masks) works again.
During training, the response mask DataProto is unioned into the batch before actor updates, preventing KeyErrors when the actor expects response_mask.
* recipe/prime/prime_dp_rm.py
Reward-model forward handling now normalizes outputs that can be either ModelOutput or tuples, so fused/non‑fused kernels both work.
When fused kernels are enabled, the precomputed log_probs are used; otherwise, log probabilities are recomputed from logits. This removes the AttributeError that occurred when the fast path returned tuples.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)

### CODE REVIEW

File: recipe/prime/config/prime_trainer.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ actor_rollout_ref:
   rollout:
     # number of responses (i.e. num sample times)
     n: 4
+    mode: sync
```
Comment (wuxibin89) [inline recipe/prime/config/prime_trainer.yaml]: sync mode is deprecated and will be removed soon.

Comment (lsy641) [reply recipe/prime/config/prime_trainer.yaml]: Thanks, it seems AgentLoopManager is called. The original codes use ActorRolloutRefWorker. I add AsyncActorRolloutRefWorker for async rollout. Because AgentLoopManager is not neccesary, I think just adding AsyncActorRolloutRefWorker is enough and overide the init_workers function in RayPRIMETrainer. Does that sound good?
